server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -303,19 +309,66 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failDecommissionedNodes(ClusterState state) {
+    private void weighAwayDecommissionedZone(ClusterState state, TimeValue timeOutForNodeDraining) {
         // this method ensures no matter what, we always exit from this function after clearing the voting config exclusion
         DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
         DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+
+        // Awareness values refers to all zones in the cluster
+        List<String> awarenessValues = forcedAwarenessAttributes.get(decommissionAttribute.attributeName());
+
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.WEIGH_AWAY, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+
+                decommissionController.setWeightForDecommissionedZone(awarenessValues, new ActionListener<>() {
+                    @Override
+                    public void onResponse(ClusterPutWRRWeightsResponse response) {
+                        clearVotingConfigExclusionAndUpdateStatus(true, true);
```

Comment:
Why are we clearing the exclusion here and updating status?

Reply:
Fixed. Thanks for pointing out.